webbpsf/distortion.py

```diff
@@ -116,29 +116,30 @@ def distort_image(hdulist_or_filename, ext=0, to_frame='sci', fill_value=0,
     # Pixel scale information
     ny, nx = hdu_list[ext].shape
     pixelscale = hdu_list[ext].header["PIXELSCL"]  # the pixel scale carries the over-sample value
-    oversamp   = hdu_list[ext].header["DET_SAMP"]  # PSF oversampling relative to detector 
 
     # Get 'sci' reference location where PSF is observed
     xsci_cen = hdu_list[ext].header["DET_X"]  # center x location in pixels ('sci')
     ysci_cen = hdu_list[ext].header["DET_Y"]  # center y location in pixels ('sci')
 
+    # Convert the PSF center point from pixels to arcseconds using pysiaf
+    xidl_cen, yidl_cen = aper.sci_to_idl(xsci_cen, ysci_cen)
+
     # ###############################################
     # Create an array of indices (in pixels) for where the PSF is located on the detector
     nx_half, ny_half = ( (nx-1)/2., (ny-1)/2. )
     xlin = np.linspace(-1*nx_half, nx_half, nx)
     ylin = np.linspace(-1*ny_half, ny_half, ny)
     xarr, yarr = np.meshgrid(xlin, ylin) 
 
-    # Convert the PSF center point from pixels to arcseconds using pysiaf
-    xidl_cen, yidl_cen = aper.sci_to_idl(xsci_cen, ysci_cen)
-
-    # Get 'idl' coords
-    xidl = xarr * pixelscale + xidl_cen
-    yidl = yarr * pixelscale + yidl_cen
-
     # ###############################################
     # Create an array of indices (in pixels) that the final data will be interpolated onto
-    xnew_cen, ynew_cen = aper.convert(xsci_cen, ysci_cen, 'sci', to_frame)
+    if to_frame=='sci':
+        xnew_cen, ynew_cen = (xsci_cen, ysci_cen)
+    else:
+        # Choose conversion function
+        func_convert = getattr(aper, f'sci_to_{to_frame}')
+        xnew_cen, ynew_cen = func_convert(xsci_cen, ysci_cen)
```

Comment:
Can you explain the motivation behind this change?  It appears to me that this code is just replicating what the `aper.convert` function already does: check for a null transformation and if so return the input x,y; otherwise look up the correct X_to_Y function and then call it.

Reply:
Ah, indeed. I had looked at the pysiaf `convert` function a long time ago and (perhaps erroneously) recall it making additional unnecessary transformations, so have always written my own set of if statements to minimize the number of calls. But looks like the logic now is much improved.